### DISCUSSION TITLE
Manually triggered MOBIKE updates.

### DISCUSSION THREAD

**Discussion by zekena2** [strongswan/strongswan#2056] [Q&A] [answered]
Can I make strongswan manually change the interface/address that's used for the SA and choose another interface/address myself not by bringing the interface down and letting strongswan choose the new route by itself?

**Comment by tobiasbrunner** [comment, marked as answer]
Besides changes to interfaces and addresses there are other triggers for roaming events. Route and routing rule updates can also trigger them (the latter has to be enabled via `charon.process_rules`, the former is enabled by default). The source address/interface is determined by the routes, however, if the current path is still available, no MOBIKE update will occur. To force a switch to the "best" path, you'll have to enable `charon.prefer_best_path`.

**Reply by zekena2** [reply]
How can I choose the "best" path "for me" without or with very minimal disruption?

**Reply by tobiasbrunner** [reply]
What do you mean?

**Reply by zekena2** [reply]
Let's imagine a situation where I have 3 interfaces one is used currently for the connection and I decided that I want to switch to another one for reasons other than the whole interface being down. Do I have to make this interface the only one up "whether through routing rules or with it bringing down the other interfaces" or is there a way to tell strongswan this is the interface I want to switch to?

**Reply by tobiasbrunner** [reply]
As I said, that's determined by the routes e.g. their metrics. strongSwan's custom route lookup does currently not consider rules, so routes from all tables are treated equally. To use rules, you have to switch to the kernel's native route lookup (as documented for `charon.process_rules` e.g. via `charon.plugins.kernel-netlink.fwmark`).

**Reply by zekena2** [reply]
It would be a very beneficial feature to manually tell strongswan to do the mobike update and decide the new address myself via vici for example.

Another question would be if there's an example on how to use routes and rules to trigger a roaming event?

**Reply by tobiasbrunner** [reply]
> It would be a very beneficial feature to manually tell strongswan to do the mobike update and decide the new address myself via vici for example.

The whole point of this is that it works automatically and reacts to environment changes without explicit user intervention.

> Another question would be if there's an example on how to use routes and rules to trigger a roaming event?

Any changes to the interfaces, addresses and routes (and rules if enabled) on the system will trigger a roam event that evaluates the paths for all established IKE_SAs and then potentially triggers a MOBIKE event if the source address changes (either because the current one isn't available anymore or there is a better path). So just update the routes/rules according to whatever it is you want to achieve.
